spring-web/src/main/java/org/springframework/http/converter/ByteArrayHttpMessageConverter.java

```diff
@@ -52,7 +52,15 @@ public boolean supports(Class<?> clazz) {
 
 	@Override
 	public byte[] readInternal(Class<? extends byte[]> clazz, HttpInputMessage inputMessage) throws IOException {
-		return inputMessage.getBody().readAllBytes();
+		long contentLength = inputMessage.getHeaders().getContentLength();
+		final int len;
+		if (contentLength >= 0 && contentLength <= Integer.MAX_VALUE) {
+			len = (int) contentLength;
+		}
+		else {
+			len = Integer.MAX_VALUE;
+		}
+		return inputMessage.getBody().readNBytes(len);
```

Comment:
Regarding readAllBytes, we read with chunks with max len 8192 (default_buffer_size). 
Do we really get better performance with this code?